x/stake/handler_sidechain.go

```diff
@@ -476,6 +480,102 @@ func handleMsgSideChainUndelegate(ctx sdk.Context, msg MsgSideChainUndelegate, k
 	return sdk.Result{Data: finishTime, Tags: tags, Events: events}
 }
 
+func handleMsgSideChainStakeMigration(ctx sdk.Context, msg MsgSideChainStakeMigration, k keeper.Keeper) sdk.Result {
+	if scCtx, err := k.ScKeeper.PrepareCtxForSideChain(ctx, k.DestChainName); err != nil {
+		return ErrInvalidSideChainId(k.Codespace()).Result()
+	} else {
+		ctx = scCtx
+	}
+
+	denom := k.BondDenom(ctx)
+	if msg.Amount.Denom != denom {
+		return ErrBadDenom(k.Codespace()).Result()
+	}
+
+	shares, sdkErr := k.ValidateUnbondAmount(ctx, msg.RefundAddress, msg.Validator, msg.Amount.Amount)
+	if sdkErr != nil {
+		return sdkErr.Result()
+	}
+
+	// unbond immediately
```

Comment:
```
	ubd, sdkErr := k.BeginUnbonding(ctx, msg.RefundAddress, msg.Validator, shares, false)
	if sdkErr != nil {
		return sdkErr.Result()
	}
	ubd, events, sdkErr := k.CompleteUnbonding(ctx, ubd.DelegatorAddr, ubd.ValidatorAddr)
	if sdkErr != nil {
		return sdkErr.Result()
	}
```
can be replaced by `UnboundDelegation`.

I think this function `UnboundDelegation` is Dylan implemented for you. 

If you think it is unnecessary, please remove the `UnboundDelegation` function.

Reply:
Done